homeassistant/util/logging.py

```diff
@@ -164,3 +164,45 @@ def wrapper(*args: Any) -> None:
                 log_exception(*args)
         wrapper_func = wrapper
     return wrapper_func
+
+
+def catch_log_coro_exception(
+        target: Coroutine[Any, Any, Any],
+        format_err: Callable[..., Any],
+        *args: Any) -> Coroutine[Any, Any, Any]:
+    """Decorate a coroutine to catch and log exceptions."""
+    def log_exception(*args: Any) -> None:
```

Comment:
please define outside of this function.

Reply:
Actually any reason it's a function and not just written out in line 187.

Reply:
Fixed!